### PR TITLE
[Merged by Bors] - Allow users to disable CA auto-generation

### LINKED ISSUE #67: CAs should not be created automagically
As a user I'd like it to be my concious choice whether a CA should be created automatically or not when the referenced Secret does not exist so that I am informed about any typos I might have in my definitions (or even just the order of applies being wrong).

### PR DESCRIPTION
**Pull request by teozkr**
## Description

Fixes #67

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: src/backend/tls.rs [hunk 1]
```diff
@@ -34,6 +34,11 @@ use super::{pod_info::Address, pod_info::PodInfo, SecretBackend, SecretBackendEr
 pub enum Error {
     #[snafu(display("failed to generate certificate key"))]
     GenerateKey { source: openssl::error::ErrorStack },
+    #[snafu(display("could not find CA, and autoGenerate is false"))]
```
Comment (sbernauer) [inline src/backend/tls.rs:37]:
```suggestion
    #[snafu(display("could not find CA in secret {secret}, and autoGenerate is false"))]
```

**Review by sbernauer — APPROVED** [review]

**Comment by sbernauer** [comment]
LGTM % comment

**Review by soenkeliebau — APPROVED** [review]
Looks good to me. 

Installed from helm and tested functionality.

**Pull request closed without merging**